app/billing_client.go

```diff
@@ -0,0 +1,310 @@
+package app
+
+import (
+	"context"
+	"errors"
+	"io"
+
+	pb "go.viam.com/api/app/v1"
+	"go.viam.com/utils/rpc"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+// UsageCostType specifies the type of usage cost.
+type UsageCostType int32
```

Comment:
[nit] This and the enum below can just be `int`s, I think (`int32` or `int64` depending on your system.) Using `int32` implies some need for a smaller int that probably is not necessary.